safesocket/safesocket_darwin.go

```diff
@@ -83,6 +86,11 @@ func localTCPPortAndTokenDarwin() (port int, token string, err error) {
 				}
 			}
 		}
+		if os.Getuid() == 0 {
+			warnAboutRootOnce.Do(func() {
```

Comment:
Done as a separate warning log because the error returned by this function is swallowed (we continue on the fallback path at https://github.com/tailscale/tailscale/blob/a1dbd56c18ccd480c72e4d7876065a52bde19acc/client/tailscale/tailscale.go#L67-L71).

Reply:
When you run `sudo`, is the `TS_MACOS_CLI_SHARED_DIR` environment set in the new environment? I always forget how that works. But I assume if you tested this, then it works as-is.

Reply:
Environment variables are not inherited when using `sudo` (unless you use `-E`), but in this case we first start running as root, then we set the environment variable, then we run the CLI (in the same process, where we read it back out), so it's not a concern.